site/en/articles/new-headless/index.md

```diff
@@ -0,0 +1,204 @@
+
+```sh
+chrome --headless=new --screenshot --window-size=412,892 https://developer.chrome.com/
+```
+
+#### `--print-to-pdf`
+
+The `--print-to-pdf` flag saves the target page as a PDF named `output.pdf` in the current working directory. Here’s an example:
+
+```sh
+chrome --headless=new --print-to-pdf https://developer.chrome.com/
+```
+
+#### `--timeout`
+
+The `--timeout` flag specifies the delay in milliseconds after which the page’s content is captured by `--dump-dom`, `--screenshot`, and `--print-to-pdf`. When neither `--timeout` nor `--virtual-time-budget` (see below) are specified, the page content is captured as soon as the page is loaded.
+
+To illustrate its use, consider [this demo page which increments, logs, and displays a counter every second](https://mathiasbynens.be/demo/time) using `setTimeout(fn, 1000)`. Here’s the relevant code:
+
+```html
+<output>0</output>
+<script>
+  const element = document.querySelector('output');
+  let counter = 0;
+  setInterval(() => {
+    counter++;
+    console.log(counter);
+    element.textContent = counter;
+  }, 1_000);
+</script>
+```
+
+After one second, the page contains “1”; after two seconds, “2”, and so on. Here’s how you’d capture the page’s state after 5 seconds and save it as a PDF:
+
+```sh
+chrome --headless=new --print-to-pdf --timeout=5000 https://mathiasbynens.be/demo/time
+```
+
+The `--timeout=5000` flag tells Chrome to wait for 5 seconds before printing the PDF. Thus, this process takes at least 5 seconds to run.
+
+#### `--virtual-time-budget`
+
+The `--virtual-time-budget` enables time travel! Well, kinda. Virtual Time acts as a “fast-forward” for any time-dependent code (e.g. `setTimeout`/`setInterval`). It forces the browser to execute any of the page’s code as fast as possible while making the page believe that the time actually goes by.
+
+`--virtual-time-budget` is most commonly used as a replacement for `--timeout`, like this:
+
+```sh
+chrome --headless=new --print-to-pdf --virtual-time-budget=5000 https://mathiasbynens.be/demo/time
+```
+
+This produces the same result as the previous example with `--timeout`. The difference is that with `--virtual-time-budget` the process takes very little real time — just about the same time as if there was no timeout specified at all. The difference becomes more obvious with larger values:
+
+```sh
+chrome --headless=new --print-to-pdf --virtual-time-budget=42000 https://mathiasbynens.be/demo/time
+```
+
+With `--timeout=42000` it would take at least 42 seconds before the PDF gets printed. With `--virtual-time-budget=42000` it barely takes longer than the previous example.
+
+## Debugging
+
+Since Chrome is effectively invisible in Headless mode, it might sound tricky to figure out what’s going on in case of issues. Luckily, it’s possible to debug Headless Chrome in a way that’s very similar to headful Chrome. The trick is to launch Chrome in Headless mode with the `--remote-debugging-port` command-line flag.
```

Comment:
```suggestion
Because Chrome is effectively invisible in Headless mode, it might sound tricky to figure out what’s going on in case of issues. Luckily, it’s possible to debug Headless Chrome in a way that’s very similar to headful Chrome. The trick is to launch Chrome in Headless mode with the `--remote-debugging-port` command-line flag.
```